### PR TITLE
Updated pull request based on branch V4.1.0 from solid/node-solid-server

### PR DESCRIPTION
**Pull request by smalinin**

### CODE REVIEW

**Comment by RubenVerborgh** [comment]
@smalinin Thanks, could you summarize the changes?

Note: lets make this a squash merge, because of the difficult history tail (with "sync" commits instead of merge commits).

**Comment by RubenVerborgh** [comment]
Also, the build is currently broken because of

```
/home/travis/build/solid/node-solid-server/lib/requests/login-request.js:175:11: Infix operators must be spaced.
```

**Comment by smalinin** [comment]
It looks, that only the next fixes were added

File: lib/create-app.js [hunk 1]
```diff
@@ -169,7 +169,7 @@ function initWebId (argv, app, ldp) {
       // without permission by including the credentials set by the Solid server.
       const origin = req.headers.origin
       const userId = req.session.userId
-      if (!argv.host.allowsSessionFor(userId, origin)) {
+      if (req.path !== '/logout' && req.path !== '/goodbye' && !argv.host.allowsSessionFor(userId, origin)) {
```
Comment (dmitrizagidulin) [inline lib/create-app.js]: Why this addition?

File: static/databrowser.html [hunk 2]
```diff
@@ -9,25 +9,29 @@
...
+    // tabulator = {}
+
+    window.document.title = 'Loading ' + uri
+    var kb = UI.store
+    var subject = kb.sym(uri)
+    var outliner = panes.getOutliner(document)
+    outliner.GotoSubject(subject, true, undefined, true, undefined);
 
-        // Authenticate the user
-        UI.authn.checkUser()
```
[the remark this answers is not in the document]
Comment (smalinin) [reply static/databrowser.html]: I have fixed the above.

File: lib/create-app.js (hunk 1 above)
Comment (smalinin) [reply lib/create-app.js]: The Logout didn't work on WebID-OIDC server, I added request to solid-auth-client

Comment (dmitrizagidulin) [reply lib/create-app.js]: Makes sense. Let's remove this change, though, the logout cookie behavior will need to be handled separately, elsewhere.

Comment (smalinin) [reply lib/create-app.js]: I have fixed chmods, may be it was changed my Samba server, it was changed for .acl som another files.

File: lib/models/account-template.js [hunk 3]
```diff
@@ -92,7 +90,8 @@ class AccountTemplate {
       tlsCertWebid: userAccount.tlsCertificateWebid,
       tlsCertModulus: userAccount.tlsCertificateModulus,
       tlsCertExponent: userAccount.tlsCertificateExponent,
-      addOwlSameAs
+      addOwlSameAs,
+      serverUri: accountManager.host.serverUri
```
Comment (dmitrizagidulin) [inline lib/models/account-template.js]: If the `accountManager` is only being used to get the `host` instance, maybe just pass in `host` directly? Instead of the full `accountManager`?

File: lib/requests/login-request.js [hunk 4]
```diff
@@ -164,37 +164,42 @@ class LoginRequest extends AuthRequest {
    * @return {string}
    */
   postLoginUrl (validUser) {
-    if (this.clientId) {
+    let uri
+
+    if (this.authQueryParams['client_id']) {
```
Comment (dmitrizagidulin) [inline lib/requests/login-request.js]: Why the change from `this.clientId` to `this.authQueryParams['client_id']` here? Was `clientId` not being properly initialized?

File: lib/create-app.js (hunk 1 above)
Comment (smalinin) [reply lib/create-app.js]: Removed handle for `/logout` and `/goodbye`

File: lib/requests/login-request.js (hunk 4 above)
Comment (smalinin) [reply lib/requests/login-request.js]: Fixed

File: lib/models/account-template.js (hunk 3 above)
Comment (smalinin) [reply lib/models/account-template.js]: Fixed

**Comment by dmitrizagidulin** [comment]
@smalinin Thanks for the fixes! There's still 3 .acl files with `chmod` changes, but overall 👍 from me!

**Comment by smalinin** [comment]
@dmitrizagidulin Ok, I see, it looks that in my prev PR to Node-solid-server, I changed chmod for these files to `rwxrwxr-x` mode, and this was missed in prev PR review stage. Now I set chmod to `rw-rw-r--` for all .acl files, that is the properly value.

**Comment by dmitrizagidulin** [comment]
No prob. Thanks again!

**Comment by RubenVerborgh** [comment]
Reminder that we should squash this into one commit, as 41d9bb5691a037c2942dacd1a0ffe2ecef89fcbb is problematic (not a merge commit, will mess with our history).

**Comment by kidehen** [comment]
@RubenVerborgh @dmitrizagidulin ,

What does "Reminder that we should squash this into one commit, as 41d9bb5 is problematic (not a merge commit, will mess with our history)." mean in regards to:

1. Actions Items?
1. Action item ownership?

**Comment by RubenVerborgh** [comment]
> Actions Items?

Whoever merges this, should squash merge, not create a merge commit.

> Action item ownership?

The merger.

**Comment by RubenVerborgh** [comment]
Assigning this to @kjetilk for now, as a reminder we need to look into this.

**Comment by RubenVerborgh** [comment]
@smalinin @kidehen We have discussed this, and @justinwb will be following up on this PR with you. Thanks for your patience.

**Comment by smalinin** [comment]
@RubenVerborgh Basically I could create for from current state on solid:v4.1.0 , apply all changes above and create a new PR(it is not problem). Will it be better for you ?

**Comment by RubenVerborgh** [comment]
@smalinin That would help. Note that a new PR is not needed for that, you can just force push to the branch (but I don't mind either).

**Comment by kidehen** [comment]
@justinwb,

What's happening here?

**Comment by justinwb** [comment]
@kidehen just left you a note on Gitter. Summation - we’ve got it on the backlog to merge as soon as we wrap a couple more priority items for the 5.0 release. One thing that would help speed things up a ton when we merge is if the contributors in that pull can go through and add as many comments as possible ahead of time so we’re not going back and forth with a ton of questions

**Comment by smalinin** [comment]
@RubenVerborgh I have recreated PR https://github.com/solid/node-solid-server/pull/728  with fresh form and collect all changes to one commit.

**Comment by RubenVerborgh** [comment]
Note, the branch has been renamed into feature/webid-delegation, since this will not be part of v4.1.

**Comment by RubenVerborgh** [comment]
Following up in #728.

File: lib/create-app.js (hunk 1 above)
Comment (RubenVerborgh) [reply lib/create-app.js]: This addresses the symptoms, but not the cause, which should be addressed by https://github.com/solid/node-solid-server/pull/793
